client/my-sites/customer-home/cards/launchpad/use-celebrate-launch-modal.ts

```diff
@@ -1,38 +0,0 @@
-import { updateLaunchpadSettings } from '@automattic/data-stores';
-import { useState } from 'react';
-import { useDispatch } from 'calypso/state';
-import { requestSite } from 'calypso/state/sites/actions';
-
-export function useCelebrateLaunchModal( siteId: number, layout: { refetch: () => void } | null ) {
-	const [ isOpen, setIsOpen ] = useState( false );
-	const dispatch = useDispatch();
-
-	const setModalIsOpen = ( isOpen: boolean ) => {
-		setIsOpen( isOpen );
-
-		if ( isOpen ) {
-			// Site launched, update site data
-			dispatch( requestSite( siteId ) );
-		} else {
-			// Modal closed, update the launchpad data/checklist
-			layout?.refetch();
-		}
-	};
-
-	const handleSiteLaunched = ( isWpcomAtomic: boolean ) => {
-		setModalIsOpen( true );
-		// currently the action to update site_launch status on atomic doesn't fire
-		// this is a workaround until that is fixed
-		if ( isWpcomAtomic ) {
-			updateLaunchpadSettings( siteId, {
-				checklist_statuses: { site_launched: true },
-			} );
-		}
-	};
```

Comment:
It's weird to think we had such logic inside a visual component, but I do get why.
Now, we moved this requestSite data, layout refetch, and updating launchpad settings [to the data layer](https://github.com/Automattic/wp-calypso/pull/104868/files#diff-5da04cda4c327ddd0fb5469cd850ebd4d763dec3f16ba9782753e705045270ecR31-R43) that gets triggered after the launch site request succeeds.